test/1.4/structure/T_trait.dml

```diff
@@ -52,27 +52,46 @@ port p[i < 2] {
     }
 }
 
+// The way param pointers are stored, param values must appear on addresses
+// aligned by two; without the alignment logic, references to one of the below
+// params will break.
```

Comment:
A better way to test that would be to check the addresses directly. That's not possible in DML directly without `extern`ing  the vtable (and `extern typedef`ing its type), which can be fragile. Your opinion?

Reply:
I'm not sure that would make the test any more complete. We still have the problem that DMLC might emit code that just happens to place these variables on even addresses. I placed the declarations side-by-side and alphabetically adjacent, in the hope that any reasonable implementation would place them in adjacent memory locations. If they *do* appear in adjacent locations, then the current test is guaranteed to work. If not, the test may falsely pass on a broken compiler (because the variables will likely be auto-aligned by appearing after a large variable).